flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/reader/PulsarPartitionSplitReader.java

```diff
@@ -196,7 +197,14 @@ public void handleSplitsChanges(SplitsChange<PulsarPartitionSplit> splitsChanges
         MessageId latestConsumedId = registeredSplit.getLatestConsumedId();
 
         if (latestConsumedId != null) {
-            LOG.info("Reset subscription position by the checkpoint {}", latestConsumedId);
+            if (latestConsumedId instanceof BatchMessageIdImpl) {
```

Comment:
I prefer to use `MessageIdAdv` globally.

Reply:
make sense!

Reply:
So. I think there is no need to cast to `BatchMessageIdImpl`. Slf4j will print the log by calling the toString method. Right?

Reply:
I can see that all the message implementation implement the `MessageIdAdv` interface. Which contains all the required information for the client. I think it's more better to use MessageIdAdv instead of the `MessageId` here in the whole connector.

```
/**
 * The {@link MessageId} interface provided for advanced users.
 * <p>
 * All built-in MessageId implementations should be able to be cast to MessageIdAdv.
 * </p>
 */
```